addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/ModbusDataThingHandler.java

```diff
@@ -0,0 +1,816 @@
+                manager = bridgeHandler.getManagerRef().get();
+                childOfEndpoint = true;
+            } else {
+                @SuppressWarnings("null")
+                @NonNull
+                ModbusPollerThingHandler bridgeHandler = (@NonNull ModbusPollerThingHandler) bridge.getHandler();
+                pollTask = bridgeHandler.getPollTask();
+                if (pollTask == null) {
+                    logger.debug("Poller {} '{}' has no poll task -- configuration is changing?", bridge.getUID(),
+                            bridge.getLabel());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                            String.format("Poller %s '%s' has no poll task", bridge.getUID(), bridge.getLabel()));
+                    return;
+                }
+                slaveId = pollTask.getRequest().getUnitID();
+                slaveEndpoint = pollTask.getEndpoint();
+                manager = bridgeHandler.getManagerRef().get();
+                pollStart = pollTask.getRequest().getReference();
+                childOfEndpoint = false;
+            }
+            if (!validateAndParseReadParameters()) {
+                // status already updated to OFFLINE
+                return;
+            }
+            if (!validateAndParseWriteParameters()) {
+                // status already updated to OFFLINE
+                return;
+            }
+            if (!validateMustReadOrWrite()) {
+                // status already updated to OFFLINE
+                return;
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Throwable e) {
+            logger.error("Exception during initialization", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, String
+                    .format("Exception during initialization: %s (%s)", e.getMessage(), e.getClass().getSimpleName()));
+        } finally {
+            logger.trace("initialize() of thing {} '{}' finished", thing.getUID(), thing.getLabel());
+        }
+    }
+
+    @Override
+    public synchronized void dispose() {
+        readValueType = null;
+        writeValueType = null;
+        readTransformation = null;
+        writeTransformation = null;
+        readIndex = Optional.empty();
+        readSubIndex = Optional.empty();
+        writeStart = null;
+        pollStart = 0;
+        slaveId = 0;
+        slaveEndpoint = null;
+        manager = null;
+        pollTask = null;
+        isWriteEnabled = false;
+        isReadEnabled = false;
+        transformationOnlyInWrite = false;
+        childOfEndpoint = false;
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Thing disposed -- bridge initializing?");
+    }
+
+    @Override
+    public synchronized void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged for {}. Reseting handler", this.getThing().getUID());
+        this.dispose();
```

Comment:
The default implementation just changes the status of the thing, but not does not call dispose() or initialize().

By calling dispose+initialize, the relevant parameters and data is read from the parent bridges. For example, it might be that poller has been modified such that the data things are no longer valid (configuration error).